src/monster.cpp

```diff
@@ -482,17 +482,13 @@ void Monster::onCreatureLeave(Creature* creature)
 	//update targetList
 	if (isOpponent(creature)) {
 		removeTarget(creature);
-		if (targetList.empty()) {
+		updateIdleStatus();
+
+		if (!isSummon() && targetList.empty()) {
 			int32_t walkToSpawnRadius = g_config.getNumber(ConfigManager::DEFAULT_WALKTOSPAWNRADIUS);
 			if (walkToSpawnRadius > 0 && !Position::areInRange(position, masterPos, walkToSpawnRadius, walkToSpawnRadius)) {
```

Comment:
the Position::areInRange overload is unnecessary, we already have: `Position::areInRange<x, y, z>(pos1, pos2)` you should delete it and use the one that already exists.

Reply:
You can't use it for non const value 🙄

Reply:
are you changing the value of `DEFAULT_WALKTOSPAWNRADIUS`?
```c++
const int32_t walkToSpawnRadius = g_config.getNumber(ConfigManager::DEFAULT_WALKTOSPAWNRADIUS);
```

Reply:
Nice. Now try to compile it.